SIPS/sip-12.md

```diff
@@ -107,16 +109,29 @@ the request is looking for resolution to a domain.
 The interface for the return value of an `onNameLookup` export is:
 
 ```typescript
+type AddressResolution = {
+  protocol: string;
+  resolvedAddress: AccountAddress;
+};
+
+type DomainResolution = {
+  protocol: string;
+  resolvedDomain: string;
+};
+
 type OnNameLookupResponse =
   | {
-      resolvedAddress: AccountAddress;
-      resolvedDomain?: never;
+      resolvedAddresses: AddressResolution[];
```

Comment:
What is the use-case for this? A snap resolving for multiple types for name resolution protocols?

Reply:
A snap resolving for multiple protocols